openedx/features/edly/utils.py

```diff
@@ -1,5 +1,37 @@
 import jwt
 from django.conf import settings
+from django.contrib.sites.shortcuts import get_current_site
+from django.db.models import Q
+
+from openedx.features.edly.models import EdlySubOrganization
+
+
+def user_has_edly_organization_access(request):
+    """
+    Check if the requested URL site is allowed for the user.
+
+    This method checks if the requested URL site is in the requesting user's
+    edly sub organizations list.
+
+    Arguments:
+        request: HTTP request object
+
+    Returns:
+        bool: Returns True if User has Edly Organization Access Otherwise False.
+    """
+    if not request.user.edly_profile:
+        return False
+
+    current_site = get_current_site(request)
+    edly_sub_org = EdlySubOrganization.objects.filter(
+        Q(lms_site=current_site) |
+        Q(studio_site=current_site)
+    ).first()
+
+    if edly_sub_org and edly_sub_org.slug in request.user.edly_profile.get_linked_edly_sub_organizations:
```

Comment:
here we should mat the `edly_sub_org.slug` with sub_org_slug of `edly-user-info` cookie.

Reply:
@ziafazal since we have all the db relations in LMS/Studio that why I thought maybe we can use these relations for finding if a user has access for an organization.

@asadmanzoor93 I think in order to be consistent with other edly cloud services and to avoid db queries, we can fetch `edly-sub-org` from the `edly-user-info` cookie. we can create a new method `get_edly_sub_org_from_cookie` which will decode the cookie `decode_edly_user_info_cookie` and return the value of `edly-sub-org`. Then match that value with `edly_sub_org.slug`.

Reply:
@ziafazal @zubair-arbi feedback implemented.